counsel-gtags.el

```diff
@@ -269,16 +296,61 @@
                 "gtags" "-q" (concat "--gtagslabel=" label))))
     (set-process-sentinel
      proc
-     (lambda (p _event)
-       (when (eq (process-status p) 'exit)
-         (kill-buffer proc-buf)
-         (message "%s: creating tag files(label: %s) in %s"
-                  (if (zerop (process-exit-status p)) "Success" "Failed")
-                  label rootdir))))))
+     (counsel-gtags--make-gtags-sentinel 'create))))
+
+
+(defun counsel-gtags--real-file-name ()
+  (let ((buffile (buffer-file-name)))
+    (unless buffile
+      (error "This buffer is not related to file."))
+    (if (file-remote-p buffile)
+        (tramp-file-name-localname (tramp-dissect-file-name buffile))
+      (file-truename buffile))))
+
+(defun counsel-gtags--read-tag-directory ()
+  (let ((dir (read-directory-name "Directory tag generated: " nil nil t)))
+    ;; On Windows, "gtags d:/tmp" work, but "gtags d:/tmp/" doesn't
+    (directory-file-name (expand-file-name dir))))
+
+(defsubst counsel-gtags--how-to-update-tags ()
+  (cl-case (prefix-numeric-value current-prefix-arg)
+    (4 'entire-update)
+    (16 'generate-other-directory)
+    (otherwise 'single-update)))
+
+(defun counsel-gtags--update-tags-command (how-to)
+  (cl-case how-to
+    (entire-update '("global" "-u"))
+    (generate-other-directory (list "gtags" (counsel-gtags--read-tag-directory)))
+    (single-update (list "global" "--single-update" (counsel-gtags--real-file-name)))))
+
+(defun counsel-gtags--update-tags-p (how-to interactive-p current-time)
+  (or interactive-p
+      (and (eq how-to 'single-update)
+           (buffer-file-name)
+           (or (not counsel-gtags-update-interval-second)
+               (>= (- current-time counsel-gtags--last-update-time)
+                   counsel-gtags-update-interval-second)))))
+
+;;;###autoload
+(defun counsel-gtags-update-tags ()
+  "Update TAG file. Update All files with `C-u' prefix.
+Generate new TAG file in selected directory with `C-u C-u'"
+  (interactive)
+  (let ((how-to (counsel-gtags--how-to-update-tags))
+        (interactive-p (called-interactively-p 'interactive))
+        (current-time (float-time (current-time))))
+    (when (counsel-gtags--update-tags-p how-to interactive-p current-time)
+      (let* ((cmds (counsel-gtags--update-tags-command how-to))
+             (proc (apply #'start-file-process "counsel-gtags-update-tag" nil cmds)))
+        (if (not proc)
+            (message "Failed: %s" (string-join cmds " "))
```

Comment:
No worries - have just pushed an extra commit for this https://github.com/syohex/emacs-counsel-gtags/pull/1/commits/e9b709c8981620b0142c17b320fe8982edf28e1b